### PR TITLE
fixed a small float value of retry_backoff

### PR DESCRIPTION
**Pull request by ipakeev**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryq.dev/en/main/contributing.html).

## Description
I encountered an unexpected behavior: if I set retry_backoff to 0.0 < {value} < 1.0, all retries will run after default_retry_delay (180 seconds). 
I don't think it makes much sense to allow a floating retry_backoff value, so I propose just rounding the {value} up to 1.

### CODE REVIEW

**Review by auvipy — CHANGES_REQUESTED** [review]
can you please add additional test cases for this change?

**Comment by ipakeev** [comment]
@auvipy Added tests.

**Comment by auvipy** [comment]
it would be great to add additional integration tests

**Comment by ipakeev** [comment]
Added additional tests.

@auvipy what kind of integration tests do you expect? I've not found any integration cases in which retry_backoff is used.

File: celery/app/autoretry.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def run(*args, **kwargs):
                 if retry_backoff:
                     retry_kwargs['countdown'] = \
                         get_exponential_backoff_interval(
-                            factor=retry_backoff,
+                            factor=int(max(1.0, retry_backoff)),
```
Comment (auvipy) [inline celery/app/autoretry.py:51]: are we going to turn everything into 1 only or multiple of 1.0?

Comment (ipakeev) [reply celery/app/autoretry.py:51]: All values less then 1 turn into 1 (except zero). For the other values, the behavior remains the same. I'm using 1.0 (not 1) just for linter.

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
merging with the hope that no regression was introduced! if we got any report, please try to collaborate if needed. thanks for your contribution.

**Pull request merged**